gateway/src/main/java/com/mx/path/gateway/configuration/annotations/Connection.java

```diff
@@ -13,6 +13,13 @@
 @Target({ ElementType.FIELD, ElementType.PARAMETER })
 public @interface Connection {
 
+  /**
+   * The connection name to be bound from gateway configuration.
+   */
   String value();
 
+  /**
+   * Indicates whether the connection is optionally required to be configured in the gateway configuration. Default is false.
+   */
+  boolean optional() default false;
```

Comment:
I think we use `required` in other places. I think that would be clearer. Sorry, I think the rough idea came out with an optional attribute. 😆

Reply:
Default to `true`

Reply:
After some discussion we think 'optional' works as well.

Reply:
Just to be clear: I'm sticking with @Connection(value = "name", optional = true) to declare a connection may come in null.

@Connection("name") will throw an exception if the connection is not configured.

This means that an accessor that doesn't specify a connection is optional won't have to null-check the connection in the code.

Reply:
Agreed.